jcg_testadapter_commons/src/main/scala/JavaTestAdapter.scala

```diff
@@ -0,0 +1,14 @@
+trait JavaTestAdapter extends TestAdapter {
+    override val language: String = "java"
+
+    // Overload serializeCG for java
+    def serializeCG(algorithm: String, target: String, mainClass: String, classPath: Array[String], JDKPath: String, analyzeJDK: Boolean, outputFile: String): Long
+
+    // Override base methods
+    override def serializeCG(algorithm: String, outputDirPath: String, inputDirPath: String): Long = {
```

Comment:
What is the point of overriding these methods empty? If at all, shouldn't it rather be throwing an OperationNotSupportedException?
But in general, this might point to a design issue. Maybe these methods could take a config object (either only that or in addition to the current parameters) that can then contain language specific parameters like the classpath and JDK path for Java?

Reply:
I agree that this definitely is a design issue! I did some more refactoring, so now the serializeCG method also takes an optional adapterOptions parameter, that we can use to pass the additional information and unify the interface. 

The interfaces might become even more unified, once I implement the timeouts!